doc/source/rllib-training.rst

```diff
@@ -288,6 +288,67 @@ directly as ``compute_action()`` does:
             return res[0]  # backwards compatibility
 
 
+Training, Saving, and Testing Workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Python API allows configuring custom workflows for training RLlib agents, saving the trained agents, and testing them.
+Below is a recommended workflow leveraging ``tune.run()`` to train a PPO agent, which allows specifying configurable stopping criteria, a custom logging directory (instead of the default at ``~/ray_results``), saving the trained agent, and returning the corresponding checkpoint path. The trained agent can be loaded from the path to be tested and evaluated.
+
+.. code-block:: python
+
+	class RlHelper:
+		def __init__(self, config, save_dir):
+			self.config = config
+			self.env_class = config['env']
+			self.env_config = config['env_config']
+			self.save_dir = save_dir
+			self.agent = None
+		
+		def train(self, stop_criteria):
+			"""
+			Train an RLlib PPO agent using tune until any of the configured stopping criteria is met.
+			:param stop_criteria: Dict with stopping criteria.
+				See https://docs.ray.io/en/latest/tune/api_docs/execution.html#tune-run
+			:return: Return the path to the saved agent (checkpoint) and tune's ExperimentAnalysis object
+				See https://docs.ray.io/en/latest/tune/api_docs/analysis.html#experimentanalysis-tune-experimentanalysis
+			"""
+			analysis = ray.tune.run(ppo.PPOTrainer, config=self.config, local_dir=self.save_dir, stop=stop_criteria,
+									checkpoint_at_end=True)
+			# list of lists: one list per checkpoint; each checkpoint list contains 1st the path, 2nd the metric value
+			checkpoints = analysis.get_trial_checkpoints_paths(trial=analysis.get_best_trial('episode_reward_mean'),
+															   metric='episode_reward_mean')
+			# retrieve the checkpoint path; we only have a single checkpoint, so take the first one
+			checkpoint_path = checkpoints[0][0]
+			return checkpoint_path, analysis
+
+		def load(self, path):
+			"""
+			Load a trained RLlib agent from the specified path. Call this before testing a trained agent.
+			:param path: Path pointing to the agent's saved checkpoint (only used for RLlib agents)
+			"""
+			self.agent = ppo.PPOTrainer(config=self.config, env=self.env_class)
+			self.agent.restore(path)
+
+		def test(self):
+			"""Test trained agent for a single episode. Return the episode reward"""
+			assert self.agent is not None, "Load agent before testing"
+			
+			# instantiate env class
+			env = self.env_class(self.env_config)
+
+			# run until episode ends
+			episode_reward = 0
+			done = False
+			obs = env.reset()
+			while not done:
+				action = self.agent.compute_action(obs)
+				obs, reward, done, info = env.step(action)
+				episode_reward += reward
+
+			return episode_reward
```

Comment:
I'm not sure about this wrapper class. For one, it seems to be a thin wrapper pattern which does simplify the training, but doesn't work for more advanced use cases.

My preference would be to clarify the existing docs for how to train/restore/test with extra code snippet examples, instead of introducing this helper.

Reply:
Ok, how about I get rid of the wrapper class and split it up into the function for train & save, load, and test and use them in separate subsections of the Python API docs? Or what do you suggest? I can also just get rid of this all together and just keep the tip on logging.